### PR TITLE
feat: Integrate Custom formkit README Docs Processor

### PR DESCRIPTION
**Pull request by chrisfalaska**
# Alaska Airlines Pull Request

Integrating a custom `kitDocProcessor` to generate `auro-formkit` `README` documentation.

Because `formkit` is a monorepo, its documentation scripts require customization, unlike those found in `auro-library` that support non-monorepo components.

Additionally, due to the differences in the folder structure of the monorepo, it was considered best practice to separate concerns and create a `kitDocProcessor` file for the `auro-formkit` itself,  separate from the pre-existing `docProcessor` file for `formkit` components.

## Additional changes

- Update `README.md` to reflect the new monorepo.
- Add detailed sections in `README.md` for managing dependencies and development practices in a monorepo setup.
- Include new partials for description, additional information, development, and testing in the documentation.

### CODE REVIEW

File: docs/partials/testing.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+```
+npm run test
+```
+
+### @TODO:
+
+#### WTR & Workspaces
+
+`web-test-runner` does not support the `node:` prefix. Therefor, this does not work:
```
[the remark this answers is not in the document]
Comment (chrisfalaska) [reply docs/partials/testing.md]: Fixed

File: turbo.json [hunk 2]
```diff
@@ -45,6 +45,10 @@
     "build:docs": {
       "outputs": ["README.md", "demo/index.*", "docs/partials/*.md"]
     },
+    "@auro-formkit/build-tools#build:docs:kit": {
+      "inputs": ["../../docs/partials/**/*", "../../docs/templates/**/*", "src/kitDocProcessor.mjs"],
```
Comment (sun-mota) [inline turbo.json:49]: What directory does this path relatively direct from?

Comment (chrisfalaska) [reply turbo.json:49]: These paths are relative to where the script is run from. In this case, from `packages/build-tools/`:

https://github.com/AlaskaAirlines/auro-form/blob/cfriedberg/11/packages/build-tools/package.json#L16

Comment (sun-mota) [reply turbo.json:49]: The script is running from the root. Then should the paths to be `doc/...`?

Comment (chrisfalaska) [reply turbo.json:49]: The script in the root uses Turbo which uses the `--filter` flag to target the package script:

https://github.com/AlaskaAirlines/auro-form/blob/cfriedberg/11/package.json#L35

This allows it to take advantage of Turbo's caching and features

Comment (sun-mota) [reply turbo.json:49]: Confirmed `process.cwd()` on `npm run build:docs:kit` is `packages/build_tools` 😊

**Review by sun-mota — DISMISSED** [review]

**Review by rmenner — APPROVED** [review]

File: docs/partials/readmeAddlInfo.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+#### Best practices for dependency installation
+
+When you install a dependency in a component or package in `auro-formkit`, you should install it directly in the package that uses it. 
+
+The package's `package.json` will have every dependency that it needs. This is true for both external and internal dependencies.
+
+- External dependencies come from the `npm` registry.
+
+- Internal dependencies let you share functionality within your repository.
```
Comment (jason-capsule42) [inline docs/partials/readmeAddlInfo.md:15]: I am not sure how those line up to the three types of dependencies in the package JSON (Dependency, Peer Dep, Dev Dep). 

I assume that "external" vs. "internal" is talking about combobox needs input which is internal but input also needs icon and button which is not.

Language should explain this dynamic as other teams are not likely to get it at first.

File: docs/partials/readmeAddlInfo.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+- **Improved clarity:** It's easier to understand what a package depends on when its dependencies are listed in its `package.json`. Developers working in the repository can see at a glance what dependencies are used within the package.
+
+- **Enhanced flexibility:** In a monorepo at scale, it can be unrealistic to expect each package to use the same version of an external dependency.
+
+- **Better caching ability:** If you install too many dependencies in the root of your repository, you'll be changing the workspace root whenever you add, update, or delete a dependency, leading to unnecessary cache misses.
+
+- **Pruning unused dependencies:** When dependencies are installed in the packages that they are meant for, Turborepo can read your lockfile and remove dependencies that aren't used in the packages you need.
+
+For more information, see the [Turborepo docs](https://turbo.build/repo/docs/crafting-your-repository/managing-dependencies).
```
Comment (jason-capsule42) [inline docs/partials/readmeAddlInfo.md:27]: I am not sure if I agree that explaining why we made decisions is best done in the readme. Let's leave it for now but I would likely discuss this section further before we take the component out of beta.

File: packages/build-tools/src/kitDocProcessor.mjs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      await processContentForFile(fileConfig);
+    } catch (err) {
+      Logger.error(`Error processing ${fileConfig.identifier}: ${err.message}`);
+    }
+  }
+}
+
+processDocFiles().catch(err => {
+  Logger.error(`Failed to process docs: ${err}`);
+});
```
Comment (jason-capsule42) [inline packages/build-tools/src/kitDocProcessor.mjs:66]: You need to add a OEF line to all new files. There is a feature you can turn on in VS Code to auto add them for you in the future.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

**Review by jason-capsule42 — APPROVED** [review]

**Review by rmenner — DISMISSED** [review]

**Review by rmenner — APPROVED** [review]

**Pull request merged**